pkg/terraform/gather/ovirt/ip.go

```diff
@@ -2,13 +2,124 @@
 package ovirt
 
 import (
+	//"context"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/openshift/installer/pkg/asset/installconfig/ovirt"
 	"github.com/openshift/installer/pkg/terraform"
+	ovirtsdk4 "github.com/ovirt/go-ovirt"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	//"time"
 )
 
+const bootstrapSSHPort string = "22"
+
+func checkPortIsOpen(host string, port string) bool {
+	timeout := time.Second
+	conn, err := net.DialTimeout("tcp", net.JoinHostPort(host, port), timeout)
+	if err != nil {
+		logrus.Debugln(fmt.Sprintf("connection error: %v", err))
+	}
+	if conn != nil {
+		defer conn.Close()
+		return true
+	}
+	return false
+}
+
+func getreportedDevices(c *ovirtsdk4.Connection, vmID string) (*ovirtsdk4.ReportedDeviceSlice, error) {
+
+	vmsService := c.SystemService().VmsService()
+	//# Look up fot the vm by id:
+	vmResp, err := vmsService.VmService(vmID).Get().Send()
+	if err != nil {
+		return nil, fmt.Errorf("failed to find VM, by id %v, reason: %v", vmID, err)
+	}
+	vm := vmResp.MustVm()
+
+	// Get the reported-devices service for this vm:
+	reportedDevicesService := vmsService.VmService(vm.MustId()).ReportedDevicesService()
+
+	// Get the guest reported devices
+	reportedDeviceResp, err := reportedDevicesService.List().Send()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get reported devices list, reason: %v", err)
+	}
+	reportedDeviceSlice, _ := reportedDeviceResp.ReportedDevice()
+
+	if len(reportedDeviceSlice.Slice()) == 0 {
+		return nil, fmt.Errorf("cannot find IPs for vmId: %s", vmID)
+	}
+	return reportedDeviceSlice, nil
+}
+
+func findVirtualMachineIP(c *ovirtsdk4.Connection, moRefValue string) (string, error) {
+
+	ipAddress := ""
+
+	reportedDeviceSlice, err := getreportedDevices(c, moRefValue)
+	if err != nil {
+		return "", err
+	}
+
+	for _, reportedDevice := range reportedDeviceSlice.Slice() {
+		ips := reportedDevice.MustIps()
+		if ips != nil {
+			for _, ip := range ips.Slice() {
+				ipres := ip.MustAddress()
+				if ipres != "" {
+					if checkPortIsOpen(ipres, bootstrapSSHPort) {
+						ipAddress = ipres
```

Comment:
Why not just return from the function at this point?
```suggestion
						logrus.Debugln(fmt.Sprintf("ovirt vm id: %s , found usable bootstrap IP %s", moRefValue, ipres))
						return ipres, nil
```

Reply:
done